release name 14 character limit, resulting in flaky unit tests

Moniker can return 5 names in a row, which are longer than the 14 character release name limit.
This results in tiller install unit tests periodically failing..

### Example Failure
(Below snippet from CI build @ https://circleci.com/gh/kubernetes/helm/1455)
`2016-08-26 . | info: Candidate name "tufted-grasshopper" exceeds maximum length 14. Skipping.`
`2016-08-26 . | info: Candidate name "coiling-panther" exceeds maximum length 14. Skipping.`
`2016-08-26 . | info: Candidate name "lanky-marsupial" exceeds maximum length 14. Skipping.`
`2016-08-26 . | info: Candidate name "donating-possum" exceeds maximum length 14. Skipping.`
`2016-08-26 . | info: Candidate name "auxilliary-moth" exceeds maximum length 14. Skipping.`
`2016-08-26 . | warning: No available release names found after 5 tries`
`2016-08-26 . | Failed install prepare step: no available release name found`
`--- FAIL: TestInstallReleaseNoHooks (0.00s)`
`	release_server_test.go:279: Failed install: no available release name found`

### Potential Solutions
- Add support in moniker to specify the max length of the name
- Truncate the release name to 14 chars, whenever moniker returns one greater than 14

--- a/cmd/tiller/release_server.go
+++ b/cmd/tiller/release_server.go
@@ -333,8 +333,7 @@ func (s *releaseServer) uniqName(start string, reuse bool) (string, error) {
 		namer := moniker.New()
 		name := namer.NameSep("-")
 		if len(name) > releaseNameMaxLen {
-			log.Printf("info: Candidate name %q exceeds maximum length %d. Skipping.", name, releaseNameMaxLen)
-			continue
+			name = name[:releaseNameMaxLen]
 		}
 		if _, err := s.env.Releases.Get(name); err == driver.ErrReleaseNotFound {
 			return name, nil

